### PR TITLE
Chainsaws now have a suicide!

### PR DESCRIPTION
**Pull request by ghost**
Long ago, the four nations lived together in harmony. Then, everything changed when the Fire Nation attacked. Only the Chainsaw, master of all four elements, could stop them, but when the world needed him most, he wasn't really that good of an item. A hundred years passed and my brother and I discovered the new Chainsaw, a robust item with pretty sweet suicide acts.

### CODE REVIEW

File: code/game/objects/items/twohanded.dm [hunk 1]
```diff
@@ -500,6 +500,12 @@
 	actions_types = list(/datum/action/item_action/startchainsaw)
 	var/on = FALSE
 
+/obj/item/twohanded/required/chainsaw/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] begins to tear [user.p_their()] head off with the [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
```
Comment (Cyberboss) [inline code/game/objects/items/twohanded.dm]: remove the the before [src]

**Comment by kevinz000** [comment]
test uur code

File: code/game/objects/items/twohanded.dm [hunk 2]
```diff
@@ -500,6 +500,12 @@
...
+	var/obj/item/bodypart/head/myhead = user.get_bodypart("head")
+	myhead.dismember()
```
Comment (Cyberboss) [inline code/game/objects/items/twohanded.dm]: if(myhead) first

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Armhulen** [comment]
i tested my code, i'll fix the changes n stuff

**Comment by Armhulen** [comment]
and also this isn't done, i'm gonna add sound too

File: code/game/objects/items/twohanded.dm (hunk 1 above)
Comment (Armhulen) [reply code/game/objects/items/twohanded.dm]: will do sir!

**Review by Cyberboss — APPROVED** [review]

File: code/game/objects/items/twohanded.dm [hunk 3]
```diff
@@ -500,6 +500,13 @@
 	actions_types = list(/datum/action/item_action/startchainsaw)
 	var/on = FALSE
 
+/obj/item/twohanded/required/chainsaw/suicide_act(mob/living/carbon/user)
+	user.visible_message("<span class='suicide'>[user] begins to tear [user.p_their()] head off with [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+	var/obj/item/bodypart/head/myhead = user.get_bodypart("head")
+	if(myhead)
+		myhead.dismember()
```
Comment (GunHog) [inline code/game/objects/items/twohanded.dm]: YESH! I can tell my borging subjects to do this for EASY BRAINING!!

Comment (Armhulen) [reply code/game/objects/items/twohanded.dm]: emergent as fuck but i'm about to put a wrench in your plans buddy

Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/items/twohanded.dm]: It's a suicide tho

Comment (Armhulen) [reply code/game/objects/items/twohanded.dm]: i just mean that if it's not on, it doesn't dismember. IT'LL STILL KILL!

Comment (The-clown-known-as-Pepe) [reply code/game/objects/items/twohanded.dm]: airtank 2.0

**Comment by Armhulen** [comment]
@kevinz000 he who lives in a glass house should not throw stones!

**Comment by ExcessiveUseOfCobblestone** [comment]
@kevinz000 is irrelevant until I see his 150 hour tested tech webs IN ACTION

**Comment by Armhulen** [comment]
fugg i made a bad thing i gotta fix one second

File: code/game/objects/items/twohanded.dm [hunk 4]
```diff
@@ -500,6 +500,18 @@
...
 	var/on = FALSE
 
+/obj/item/twohanded/required/chainsaw/suicide_act(mob/living/carbon/user)
+	if(on)
+		user.visible_message("<span class='suicide'>[user] begins to tear [user.p_their()] head off with [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		playsound(src, "sound/weapons/chainsawhit.ogg", 100, 1)
+		var/obj/item/bodypart/head/myhead = user.get_bodypart("head")
+		if(myhead)
+			myhead.dismember()
+	if(!on) //this may look like shitcode but i had to, for some reason it wouldn't recognize the if(on) above
```
Comment (XDTM) [inline code/game/objects/items/twohanded.dm]: an `else` does not work?

Comment (Armhulen) [reply code/game/objects/items/twohanded.dm]: yeah, it doesn't recognize the if I promise. I have NO CLUE WHY.

Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/items/twohanded.dm]: What do you mean "doesn't recognize the If"

Comment (Armhulen) [reply code/game/objects/items/twohanded.dm]: the one way above, or something like that.

Comment (RemieRichards) [reply code/game/objects/items/twohanded.dm]: No, it will work, you're just doing it wrong.

Comment (Cyberboss) [inline code/game/objects/items/twohanded.dm]: Use an else

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Armhulen** [comment]
oh well okay that worked just fine

i wonder what was holding me up earlier

File: code/game/objects/items/twohanded.dm [hunk 5]
```diff
@@ -500,6 +500,18 @@
 	actions_types = list(/datum/action/item_action/startchainsaw)
 	var/on = FALSE
 
+/obj/item/twohanded/required/chainsaw/suicide_act(mob/living/carbon/user)
+	if(on)
+		user.visible_message("<span class='suicide'>[user] begins to tear [user.p_their()] head off with [src]! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		playsound(src, "sound/weapons/chainsawhit.ogg", 100, 1)
```
Comment (Cyberboss) [inline code/game/objects/items/twohanded.dm]: single quotes

File: code/game/objects/items/twohanded.dm [hunk 6]
```diff
@@ -500,6 +500,18 @@
...
+		var/obj/item/bodypart/head/myhead = user.get_bodypart("head")
+		if(myhead)
+			myhead.dismember()
+	else
+		user.visible_message("<span class='suicide'>[user] smashes [src] into [user.p_their()] neck, destroying [user.p_their()] esophagus! It looks like [user.p_theyre()] trying to commit suicide!</span>")
+		playsound(src, "sound/weapons/genhit1.ogg", 100, 1)
```
Comment (Cyberboss) [inline code/game/objects/items/twohanded.dm]: single quotes

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/items/twohanded.dm (hunk 6 above)
Comment (Armhulen) [reply code/game/objects/items/twohanded.dm]: Whoops

**Pull request merged**